.github/workflows/ci.yaml

```diff
@@ -123,6 +124,27 @@ jobs:
             install_method='rpm'"
       - name: Test readiness
         run: curl -k "https://${{ env.TEST_HOST_IP }}/api/readyz"
+
```

Comment:
Was not adding `if: always()` to the step enough?

Like
```
- name: Run playbook cleanup
  if: always()
  uses: dawidd6/action-ansible-playbook@v2
  with:
```

Ref: https://stackoverflow.com/questions/58858429/how-to-run-a-github-actions-step-even-if-the-previous-step-fails-while-still-f

Reply:
The downside of using `always()` is that it literally executes always, ignoring the `needs:` (in this case, the `needs:` is only used for ordering). This means it also executes in PRs for example :/